src/main/java/org/kohsuke/github/GitHubConnectorResponseErrorHandler.java

```diff
@@ -59,7 +65,25 @@ public void onError(@NotNull GitHubConnectorResponse connectorResponse) throws I
             if (connectorResponse.statusCode() == HTTP_NOT_FOUND) {
                 throw new FileNotFoundException(connectorResponse.request().url().toString());
             } else {
-                throw new HttpException(connectorResponse);
+                HttpException exception = new HttpException(connectorResponse);
+                if (connectorResponse.header(CONTENT_TYPE) != null && connectorResponse.header(CONTENT_TYPE).contains(TEXT_HTML)) {
+                    BufferedReader bufReader = new BufferedReader(new StringReader(exception.getMessage()));
+                    String line;
+                    int hardLineCap = 25;
+                    // <title> node is expected in the beginning anyway.
+                    //  This way we do not load the raw long images' Strings, which are later in the HTML code
+                    //  Regex or .contains would result in iterating the whole HTML document, if it didn't match UNICORN_TITLE
+                    while( (line=bufReader.readLine()) != null ) {
+                        if (line.trim().startsWith(UNICORN_TITLE)) {
+                            throw new ServiceDownException(exception);
+                        }
+                        hardLineCap--;
+                        if (hardLineCap <= 0) {
+                            break;
+                        }
+                    }
+                }
+                throw exception;
```

Comment:
Maybe move the service down check into a private helper for clarity. 
Also suggest checking `statusCode()` before anything else for efficiency.  
Either `statusCode() == HTTP_UNAVAILABLE` or at least `statusCode() >= HTTP_INTERNAL_ERROR`. 

https://docs.oracle.com/javase/8/docs/api/java/net/HttpURLConnection.html#HTTP_UNAVAILABLE


```
            if (connectorResponse.statusCode() == HTTP_NOT_FOUND) {
                throw new FileNotFoundException(connectorResponse.request().url().toString());
            } else if (isServiceDown(connectorResponse)) {
                throw new ServiceDownException(connectorResponse);
            } else { 
                throw new HttpException(connectorResponse);
            }
```